inspectkv/inspectkv.go

```diff
@@ -158,7 +324,79 @@ func ScanSnapshotTableData(store kv.Storage, ver kv.Version, t table.Table, star
 	}
 	defer snap.Release()
 
-	records, nextHandle, err := ScanTableData(t, snap, startHandle, limit)
+	records, nextHandle, err := ScanTableData(snap, t, startHandle, limit)
 
 	return records, nextHandle, errors.Trace(err)
 }
+
+// CompareTableData compares records and the corresponding table data one by one.
+// It returns nil if records is equal to the data that scans from table, otherwise
+// it returns an error with a different set of records.
+func CompareTableData(txn kv.Transaction, t table.Table, records []*RecordData) error {
+	var ret *DiffRetError
+
+	for _, r := range records {
+		vals, err := t.RowWithCols(txn, r.Handle, t.Cols())
+		if terror.ErrorEqual(err, kv.ErrNotExist) {
+			ret = newDiffRetError(resultNotExist, r.Handle, nil, r.Values)
+			break
+		}
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		if !reflect.DeepEqual(r.Values, vals) {
+			ret = newDiffRetError(r.Handle, r.Handle, vals, r.Values)
+			break
+		}
+	}
+	if ret != nil {
+		return errors.Trace(ret)
+	}
+
+	startKey := t.RecordKey(0, nil)
+	err := t.IterRecords(txn, string(startKey), t.Cols(),
+		func(h int64, vals []interface{}, cols []*column.Col) (bool, error) {
+			for _, r := range records {
+				if r.Handle == h && reflect.DeepEqual(r.Values, vals) {
+					return true, nil
+				}
+			}
+			ret = newDiffRetError(h, resultNotExist, vals, nil)
+			return false, errors.Trace(ret)
+		})
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	return nil
+}
+
+// GetTableRecordsCount returns the total number of table records.
+func GetTableRecordsCount(txn kv.Transaction, t table.Table, startHandle int64) (int64, error) {
+	startKey := t.RecordKey(startHandle, nil)
```

Comment:
IterRecords can do it， but it do something that don't need to do， and GetTableRecordsCount is easy to use.